contracts/BullaInstaPay.sol

```diff
@@ -0,0 +1,81 @@
+pragma solidity ^0.8.7;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./interfaces/IBullaManager.sol";
+
+struct Multihash {
+    bytes32 hash;
+    uint8 hashFunction;
+    uint8 size;
+}
+
+error ZeroAddress();
+error ValueMustBeGreaterThanZero();
+error ClaimTokenNotContract();
+
+contract BullaInstantPayment {
+    using Address for address;
+    using SafeERC20 for IERC20;
+
+    address public bullaManager;
+
+    constructor(address _bullaManager) {
+        bullaManager = _bullaManager;
+    }
+
+    event InstantPayment (
+        address bullaManager,
+        address indexed from,
+        address indexed to,
+        uint256 amount,
+        address tokenAddress,
+        string description,
+        string[] tags,
+        uint256 blocktime,
+        Multihash attachment
```

Comment:
since we don't need to pack these things into storage, let's actually emit a string off the event

Reply:
Do you want to accept a string as a parameter as well then?

Reply:
yes, so replace the Multihash with `string ipfsHash`